Lib/_collections_abc.py

```diff
@@ -409,6 +418,94 @@ def __subclasshook__(cls, C):
         return NotImplemented
 
 
+class _PosArgsGenericAlias(GenericAlias):
+    """ Internal class specifically to represent positional arguments in
+    ``_CallableGenericAlias``.
+    """
+    def __repr__(self):
+        return f"{__name__}._PosArgsGenericAlias" \
+               f"[{', '.join(_type_repr(t) for t in self.__args__)}]"
+
+    def __eq__(self, other):
+        o_cls = other.__class__
+        if not (o_cls.__module__ == "typing" and o_cls.__name__
+                == "_GenericAlias" or isinstance(other, GenericAlias)):
```

Comment:
```suggestion
        if not ((o_cls.__module__ == "typing"
                 and o_cls.__name__ == "_GenericAlias")
                or isinstance(other, GenericAlias)):
```
(Adding redundant parentheses to clarify the grouping of and/or, a common confusion.)

In addition, consider rearranging the logic to
```
if <condition> or <condition>:
    return <do the comparison>
return NotImplemented
```